### PR TITLE
Add multiGPU support for experiments and examples

### PR DESCRIPTION
**Pull request by Timoeller**
As easy as adding:
self.prediction_heads = nn.ModuleList([ph.to(device) for ph in prediction_heads])

Eval mode should also work now for prediction heads.

### CODE REVIEW

File: examples/doc_classification.py [hunk 1]
```diff
@@ -26,7 +24,7 @@
 ########## Settings
 ##########################
 set_all_seeds(seed=42)
-device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
+device, n_gpu = initialize_device_settings(use_cuda=True, local_rank=-1, fp16=False)
```
Comment (tholor) [inline examples/doc_classification.py]: I see why you are adding this here for multi-gpu support, but I think we should keep the examples rather simple. Can you make local_rank=-1 and fp16=False default args of  initialize_device_settings? Then we don't need to list them in the examples.

File: farm/modeling/adaptive_model.py [hunk 2]
```diff
@@ -40,7 +40,7 @@ def __init__(
         """
         super(AdaptiveModel, self).__init__()
         self.language_model = language_model.to(device)
-        self.prediction_heads = [ph.to(device) for ph in prediction_heads]
+        self.prediction_heads = nn.ModuleList([ph.to(device) for ph in prediction_heads])
```
Comment (tholor) [inline farm/modeling/adaptive_model.py]: Have you tested training, saving and loading for the examples after introducing this?

File: examples/doc_classification.py (hunk 1 above)
Comment (Timoeller) [reply examples/doc_classification.py]: good idea, done.

File: farm/modeling/adaptive_model.py (hunk 2 above)
Comment (Timoeller) [reply farm/modeling/adaptive_model.py]: I have tested training on all tasks.
I haven't tested saving, but this shouldn't impact saving at all.
I loaded a QA model that was saved with the old code and it worked. Haven't tested the other models.
